certification/internal/policy/operator/deployable_by_olm.go

```diff
@@ -151,15 +151,19 @@ func (p *DeployableByOlmCheck) isCSVReady(installedCSV string, operatorData Oper
 	go func() {
 		defer close(csvReadyDone)
 		for {
+			if ctx.Err() != nil {
+				return
+			}
+			time.Sleep(2 * time.Second)
```

Comment:
It's still possible (though less likely) for the timeout to occur after this period of time has elapsed, correct? In which case we'll still panic further down in the logic?

Reply:
The. root cause of the issue was not the concurrency, but the goroutine never terminates in the case of context timeout. I will move the if block to after the time.Sleep; but I don't think there is no easy solution for the edge case where the context is timed out while the rest of the for loop is running.

Reply:
The context itself doesn't have the power to terminate. It only informs that the timeout occurred. It's up to the goroutine itself (but not via ctx.Err(), but rather via <- ctx.Done() within the goroutine) to halt itself.